internal/registry/registry.go

```diff
@@ -0,0 +1,242 @@
+package registry
```

Comment:
I think we shouldn't merge it without UTs

Reply:
this is a hard flow to check in unit-test: you need to create repo and images. not worth it for unit-test. we also did not have it in V1

Reply:
Ack. I have talked with @qbarrand and this package is probably a good candidate for subsystem-tests when we have the occasion.

Reply:
agree